admin/authentication/middleware.js

```diff
@@ -2,18 +2,20 @@
 
 const logger = require('../services/log.service').getLogger()
 
-function isAuthenticated (role) {
+function isAuthenticated (arg) {
   return function (req, res, next) {
     if (req.isAuthenticated()) {
+      let roles = typeof arg === 'string' && arg.length > 1
+        ? [arg] : arg
       let userRole
       if (req.user) {
         userRole = ((req.user).role || {})
       }
-      logger.debug(`checking authorisation on ${req.url} for role:${role} against userRole:${userRole}`)
-      if (role === undefined || (role !== undefined && userRole && role === userRole)) {
+      logger.debug(`checking authorisation on ${req.url} for roles:${roles && roles.join()} against userRole:${userRole}`)
+      if (roles === undefined || (Array.isArray(roles) && roles.some(r => r === userRole))) {
```

Comment:
roles will now always be an array so this can be simplified.  userRole should be a valid role or undefined.

Reply:
ramda contains?

Reply:
@jon-shipley the undefined check is now removed.